pkg/restore/service_action_test.go

```diff
@@ -212,6 +215,7 @@ func TestServiceActionExecute(t *testing.T) {
 						{
 							NodePort: 8080,
 						},
+						{},
```

Comment:
What is this for?

Reply:
We preserve the `NodePort: 8080` on restore because the last applied config has an entry for it, but the `NodePort: 9090` should be removed, since there is no corresponding config entry. The nodeport removal code doesn't remove entire port entries, just the nodeport element. In a real cluster, the nodeport element is just one attribute of the port. Look at a service yaml output in a real cluster for more context. Also, note the other tests in this file to see examples where there are multiple attributes and just nodeport is removed.

Reply:
I added this to the test because the bug in the code was removing or preserving *all* nodeports based on the presence or absence of any unnamed node ports, rather than just preserving the specific unnamed ports found in the config.